### PR TITLE
setup for task 7 - post comment

### PR DESCRIPTION
**Pull request by AudeNgD**
Post comment by article id with associated tests

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -91,7 +90,47 @@ describe("app", () => {
         });
     });
   });
-
+  describe("/api/articles/:article_id/comments", () => {
+    test("POST /api/articles/1/comments", () => {
+      const newComment = {
```
Comment (nc-roisin-oreilly) [inline __tests__/app.tests.js:95]: nicely arranged, your test is clear and easy to read

File: __tests__/app.tests.js [hunk 2]
```diff
@@ -91,7 +90,47 @@ describe("app", () => {
...
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({ body }) => {
+          expect(body.comment).toEqual("test comment");
+        });
+    });
+    test("400 - author is not a user", () => {
```
Comment (nc-roisin-oreilly) [inline __tests__/app.tests.js:107]: I feel like this should be a 404, but if you have strong feelings either way let me know

File: __tests__/app.tests.js [hunk 3]
```diff
@@ -91,7 +90,47 @@ describe("app", () => {
...
+      };
+      return request(app)
+        .post("/api/articles/9999/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toEqual("article not found");
+        });
+    });
+  });
```
Comment (nc-roisin-oreilly) [inline __tests__/app.tests.js:133]: I'd also like to see a test for if the posted comment object doesn't have the keys of body or author, and a 400 test for if the id is invalid (/api/articles/banana/comments)

File: models/articles.models.js [hunk 4]
```diff
@@ -21,7 +25,22 @@ exports.fetchAllArticles = () => {
...
   return db.query(queryString).then(({ rows }) => {
-    console.log(rows, "<<here in model");
     return rows;
   });
 };
+
+exports.addCommentByArticleId = (articleId, { body, author }) => {
+  //do the checks -> author is user, article id exists in db
+  return Promise.all([
+    checkAuthorExists(author),
```
Comment (nc-roisin-oreilly) [inline models/articles.models.js:35]: really nicely done

File: utils/check-exists.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const db = require("../db/connection");
```
Comment (nc-roisin-oreilly) [inline utils/check-exists.js:1]: excellent work on these functions

**Review by nc-roisin-oreilly — COMMENTED** [review]
really great work on this ticket, Aude. I've left some comments for 2 more tests I'd like you to add, but otherwise your code and your tests are excellent, keep it up!

File: __tests__/app.tests.js (hunk 2 above)
Comment (AudeNgD) [reply __tests__/app.tests.js:107]: Refactored - changed to 404 "user not found"

File: __tests__/app.tests.js (hunk 3 above)
Comment (AudeNgD) [reply __tests__/app.tests.js:133]: Refactored code and added the tests

**Pull request merged**